plugins/cost-insights/src/types/Alert.ts

```diff
@@ -13,23 +13,147 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+import { ForwardRefExoticComponent, RefAttributes } from 'react';
 import { ChangeStatistic } from './ChangeStatistic';
+import { Duration } from './Duration';
 import { Maybe } from './Maybe';
 
 /**
  * Generic alert type with required fields for display. The `element` field will be rendered in
  * the Cost Insights "Action Items" section. This should use data fetched in the CostInsightsApi
  * implementation to render an InfoCard or other visualization.
+ *
+ * The alert type exposes hooks which can be used to enable and access various events,
+ * such as when a user dismisses or snoozes an alert. Default forms and buttons
+ * will be rendered if a hook is defined.
+ *
+ * Each default form can be overridden with a custom component. It must be implemented using
+ * React.forwardRef. See https://reactjs.org/docs/forwarding-refs
+ *
+ * Errors thrown within hooks will generate a snackbar error notification.
  */
+
 export type Alert = {
   title: string;
   subtitle: string;
+  element?: JSX.Element;
+  status?: AlertStatus;
   url?: string;
   buttonText?: string; // Default: View Instructions
-  element: JSX.Element;
+  SnoozeForm?: Maybe<AlertForm>;
+  AcceptForm?: Maybe<AlertForm>;
+  DismissForm?: Maybe<AlertForm>;
+  onSnoozed?(options: AlertOptions): Promise<Alert[]>;
+  onAccepted?(options: AlertOptions): Promise<Alert[]>;
+  onDismissed?(options: AlertOptions): Promise<Alert[]>;
+};
+
+export type AlertForm<
+  A extends Alert = any,
+  Data = any
+> = ForwardRefExoticComponent<
+  AlertFormProps<A, Data> & RefAttributes<HTMLFormElement>
+>;
+
+export interface AlertOptions<T = any> {
+  data: T;
+  group: string;
+}
+
+/**
+ * Default snooze form intervals are expressed using an ISO 8601 repeating interval string.
+ * For example, R1/P7D/2020-09-02 for 1 week or R1/P30D/2020-09-02 for 1 month.
+ *
+ * For example, if a user dismisses an alert on Monday January 01 for 1 week,
+ * it can be re-served on Monday, January 08. 7 calendar days from January 02,
+ * inclusive of the last day.
+ *
+ * https://en.wikipedia.org/wiki/ISO_8601#Repeating_intervals
+ */
+export interface AlertSnoozeFormData {
+  intervals: string;
+}
+
+export interface AlertDismissFormData {
+  other: Maybe<string>;
+  reason: AlertDismissReason;
+  feedback: Maybe<string>;
+}
+
+// TODO: Convert enum to literal
+export enum AlertStatus {
+  Snoozed = 'snoozed',
+  Accepted = 'accepted',
+  Dismissed = 'dismissed',
+}
```

Comment:
I would prefer to use [string literals](https://www.typescriptlang.org/docs/handbook/literal-types.html) instead of enums but it doesn't look like the backstage TS version supports them yet.